### PR TITLE
WIP: Orderbook & Dune Data Aggregation

### PR DESCRIPTION
**Pull request by bh2smith**
Experimenting and looking into options for aggregating data between Dune and our Orderbook. 

- We use both `sqlalchemy` and `pandas` showing syntactically what it looks like to make basic and advanced queries in each.
- We aggregate the results of querying both dune and orderbook by joining on a common column to prepare for deriving analytics from both sources. 

This is still very much a work in progress and more meant to display some options.

### CODE REVIEW

**Comment by fleupold** [comment]
Nice!

If I understand this PR correctly it shows a way of how to load records from the pg database into a python script. This mean - in combination with the existing infra - we could load data from Dune and pg into memory and operate on it using python.

One approach that I'd like to see discussed is "moving" data from Dune and pg into a joint (temporary db) on which we can then perform arbitrary SQL logic on (because for some things SQL is just more effective than writing imperative code).

**Comment by bh2smith** [comment]
Hey @gentrexha , it looks like your PR got merged into here instead of the probably intended main branch. Given that this is just a miscellaneous project and not related to any production code, I'd say we can just merge this into main. So you've got your new thing there when you want it.

**Comment by gentrexha** [comment]
> Hey @gentrexha , it looks like your PR got merged into here instead of the probably intended main branch. Given that this is just a miscellaneous project and not related to any production code, I'd say we can just merge this into main. So you've got your new thing there when you want it.

It was intended, as I wasn't sure if this miscellaneous stuff in the `orderbook-db` branch was supposed to be part of the `main` branch. Thanks!

File: src/gas_saved.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+from src.db.pg_client import pg_engine
+
+
+@click.command()
+@click.option(
+    "--batch_tx_hash",
+    help="The transaction hash of the batch you want to see the % gas saved.",
+    type=str
+)
```
Comment (bh2smith) [inline src/gas_saved.py:16]: @gentrexha - what is this decorator for here?

Comment (gentrexha) [reply src/gas_saved.py:16]: It's for the command line interface. Ref: https://click.palletsprojects.com/en/8.1.x/

Comment (bh2smith) [reply src/gas_saved.py:16]: Cool stuff!

**Pull request merged**